lib/chef/provider/aws_ebs_volume.rb

```diff
@@ -47,13 +61,30 @@ def update_aws_object(volume)
   end
 
   def destroy_aws_object(volume)
-    converge_by "delete EBS volume #{new_resource.name} in #{region}" do
-      volume.delete
+    status = volume.status
+    case status
```

Comment:
This can be DRYed to
```ruby
if status == :in_use
  attachment = volume.attachments.first
  detach(volume, :instance => attachment.instance, :device => attachment.device)
end
delete(volume)
```